app/src/main/java/com/mobileasone/dagger2workshop/data/local/NotesLocalMemoryRepository.kt

```diff
@@ -4,12 +4,13 @@ import android.support.annotation.VisibleForTesting
 import com.mobileasone.dagger2workshop.data.network.NotesServiceApi
 import com.mobileasone.dagger2workshop.domain.Note
 import com.mobileasone.dagger2workshop.domain.repositories.NotesRepository
+import javax.inject.Inject
 
 /**
  * Concrete implementation to load notes from the local data source.
  */
 class NotesLocalMemoryRepository
-private constructor(private val notesServiceApi: NotesServiceApi) : NotesRepository {
+@Inject constructor(private val notesServiceApi: NotesServiceApi) : NotesRepository {
```

Comment:
By using @Inject annotation in the constructors, Dagger will know the way to create objects of NotesLocalMemoryRepository